DOC: unify the docs for np.transpose and ndarray.transpose

SabrinaSimao opened the pull request:
Added a cross-link between` np.transpose` and `ndarray.transpose` like the one in` ndarray.sort` and `np.sort`

(this is a partial fix to https://github.com/numpy/numpy/issues/16584)

SabrinaSimao committed:
unify the docs for np.transpose and ndarray.transpose

eric-wieser commented on numpy/core/_add_newdocs.py:
```diff
@@ -4027,6 +4029,7 @@
 
     See Also
     --------
+    ndarray.transpose : Method to reverse or permute the axes of an array.
```
```suggestion
    transpose : Equivalent function
```

eric-wieser commented on numpy/core/fromnumeric.py:
```diff
@@ -624,6 +624,7 @@ def transpose(a, axes=None):
 
     See Also
     --------
+    ndarray.transpose : Method to reverse or permute the axes of an array.
```
```suggestion
    ndarray.transpose : Equivalent method
```

eric-wieser commented on numpy/core/_add_newdocs.py:
```diff
@@ -3998,6 +3998,8 @@
 
     Returns a view of the array with axes transposed.
 
+    Refer to `numpy.transpose` for full documentation.
```
Are the docs for `transpose` clearer than these ones?

SabrinaSimao replied:
Not necessarly clearer, it has different informations in Notes section.
Although, ndarray.transpose has a better example in how to convert 1-D array into 2D

SabrinaSimao replied:
I could change "for full docuentation" to "for more documentation" perhaps

SabrinaSimao replied:
I feel like the `ndarray.transpose` has more details, and it's also called by `transpose`, so i changed the way the link between them was being used

Commit:
DOC: changed the way the link is being used from transpose to ndarray.transpose

I feel like the `ndarray.transpose` has more details, and it's also called by `transpose`, so i changed the way the link between them was being used

rossbar commented:
What do you think @eric-wieser , is this sufficient to close #16584, or would a more thorough "merging" of the two docstrings be preferrable? For example, moving the non-redundant `Notes`/`Examples` from `numpy.transpose` to `numpy.ndarray.transpose` so that `ndarray.transpose` has the "main" docstring and `numpy.transpose` behaves more like a stub that simply links to the main one.

IMO centralizing is nice, as then the user has a clear-cut place to find *all* of the information about the function/method, though the cross-linking is certainly an improvement.

charris commented:
close/reopen

melissawm commented:
@eric-wieser are there any blockers to be resolved here?

eric-wieser commented:
I don't know if this is sufficient to close the linked issue, but at a glance it seems to at least be an incremental improvement, so is fine to merge.

melissawm commented:
@SabrinaSimao I am merging this as-is, but there is still work left to do if you are interested. Thanks!
